src/Components/Topic.jsx

```diff
@@ -0,0 +1,90 @@
+import { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
+import { getTopics, getArticles } from "./api";
+import ArticleList from "./ArticleList";
+import "./Topic.css";
+
+export default function Topic() {
+  const { topic: initialTopic } = useParams();
+  const [topics, setTopics] = useState([]);
+  const [currentTopic, setCurrentTopic] = useState(initialTopic || "");
+  const [articles, setArticles] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsLoading(true);
+    setIsError(false);
+
+    getTopics()
+      .then((topicsData) => {
+        setTopics(topicsData);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.error("Error loading topics: ", err);
+        setIsLoading(false);
+        setIsError(true);
+      });
+
+    if (currentTopic) {
```

Comment:
You've made a `getArticles` function flexible enough to give you the desired behaviour if the topic is defined or not, so you could get rid of this if statement which sometimes provides the topic to `getArticles` and sometimes doesn't.